### ISSUE TITLE
actions column

### ISSUE THREAD

**Issue by efaux** [free-jqgrid/jqGrid#301]
I am using json datatype to create a grid and it displays fine no problem.  Now if I add action column to the grid the column headers are fine but the first column data is missing and the rest of the columns get shifted to the left. here are images of what the grid looks like.

Without action column
![image](https://cloud.githubusercontent.com/assets/25874649/24073758/14185d5e-0bd3-11e7-8610-db29d7703b57.png)

With action column
![image](https://cloud.githubusercontent.com/assets/25874649/24073773/3cbbdccc-0bd3-11e7-8580-54ce26551d82.png)

the json data seems to be fine. 
```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
	<meta http-equiv="X-UA-Compatible" content="IE=edge">
	<meta name="viewport" content="width=device-width, initial-scale=1">
	<title>Demonstrats loadFilterDefaults:true option</title>
	<meta name="author" content="Oleg Kiriljuk">
	<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.5.0/css/font-awesome.min.css">
	<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/css/bootstrap.min.css">
	<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/css/bootstrap-theme.min.css">
	<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/bootstrap-datepicker/1.6.0/css/bootstrap-datepicker.min.css">
	<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/free-jqgrid/4.13.1/css/ui.jqgrid.min.css">
	<style>
		.ui-datepicker { font-size: 76.39%; }
	</style>
	<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.3/jquery.min.js"></script>
	<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/js/bootstrap.min.js"></script>
	<script src="https://cdnjs.cloudflare.com/ajax/libs/bootstrap-datepicker/1.6.0/js/bootstrap-datepicker.min.js"></script>
	<script>
		$.jgrid = $.jgrid || {};
		$.jgrid.no_legacy_api = true;
	</script>
	<script src="https://cdnjs.cloudflare.com/ajax/libs/free-jqgrid/4.13.1/js/jquery.jqgrid.min.js"></script>
    <style>
	 #wrapper {
		 margin: 20px 80px 0 80px;
	 }
	  .ui-jqgrid.ui-jqgrid-bootstrap {
	border: 1px solid #003380;
}
.ui-jqgrid.ui-jqgrid-bootstrap .ui-jqgrid-caption {
    background-color: #e6f0ff;
}
.ui-jqgrid.ui-jqgrid-bootstrap .ui-jqgrid-hdiv {
    background-color: #cce0ff;  /*  Header color*/  
}
 </style>
    
</head>
<script>
    //<![CDATA[
    $(function () {
        $("#grid1b").jqGrid({
            //
             url: 'url',
			  datatype: "json",
			colNames: ["","ID", "Cust ID", "Customer Name", "Address", "Address 2", "Phone"],
			  colModel: [
        		{ name: "act", template: "actions" },
				{ name: "id", align: "center", width: 40, hidden: true },
                { name: "custid", width: 250, editable: false },
                { name: "custname", width: 250 },
                { name: "addr", width: 200, search: false },
                { name: "addr2", width: 150, search: false },
                { name: "phone", width: 150}
            ],
            cmTemplate: { autoResizable: true },
			//cellurl: '../scripts/editing.php',
			//mtype: "post",
			viewrecords: true,
			autoresizeOnLoad: true,
			guiStyle: "bootstrap",
            iconSet: "fontAwesome",
			//shrinkToFit: false,
			altRows: true,
			gridview:true,
			//width: $(document).width(),
			//autowidth: true,
            idPrefix: "gb1_",
            rownumbers: false,
			pager: true,
            sortname: "custid",
            sortorder: "asc",
            caption: "The grid",
			formEditing: {
               savekey: [true, 13],
               closeOnEscape: true,
               closeAfterAdd: true,
               closeAfterEdit: true
            },
			inlineEditing: {
				keys: true
			},
            singleSelectClickMode: "selectonly", // optional setting
			beforeSelectRow: function (rowid) {
				var $self = $(this), i,
					// savedRows array is not empty if some row is in inline editing mode
					savedRows = $self.jqGrid("getGridParam", "savedRow");
				for (i = 0; i < savedRows.length; i++) {
					if (savedRows[0].id !== rowid) {
						// save currently editing row
						// one can replace saveRow to restoreRow in the next line
						$self.jqGrid("saveRow", savedRows[i].id);
					}
				}
				return savedRows.length === 0; // allow selection if saving successful
			},
            ondblClickRow: function (rowid, iRow, iCol, e) {
				//$(this).jqGrid("editGridRow", rowid);
				//$(this).jqGrid("editRow", rowid);
				$(this).jqGrid("editRow", rowid, { focusField: e.target });
			}

	      }).jqGrid("navGrid", { view: true })
		.jqGrid("inlineNav")
		.jqGrid("filterToolbar")
		.jqGrid("gridResize");
       		 //$("#grid1b").jqGrid('setFrozenColumns');
    });
    //]]>
    </script>
</head>
<body>
<div id="wrapper">
<table id="grid1b"></table>
</div>
</body>
</html>
```

**Comment by OlegKi** [comment]
Could you include the some test data, returned from the server (from `url`)? It's enough to have 2-3 rows. It's important mostly **the format** of the JSON data.

**Comment by efaux** [comment]
Here is a sample of data.

{"page":"1","total":3,"records":"57","rows":[{"id":"1","cell":["1","WILMAC  ARBYS   ","Arby's","12700
 Spine Road SW E-F5","Atlanta GA 30320","7709533300"]},{"id":"2","cell":["2","WILMAC  ATBRAP  ","Atlanta Bread Airport","12700 Spine Road Concourse A","Atlanta GA 30320",""]},{"id":"3","cell":["3","WILMAC  ATLCHO  ","Atlanta Chop House","6000 North Terminal F-8 Atrium","Atlanta GA 30230",""]}]}

Tried XML same result. Also tried with a different database and getting same result.

**Comment by OlegKi** [comment]
One can solve the problem in many ways. The first one, which I'd recommend you, is the simplest one. You need to change the format of JSON data returned from the server to the following:
```json
{
	"page": "1",
	"total": 3,
	"records": "57",
	"rows": [{
		"id": "1",
		"custid": "WILMAC ARBYS ",
		"custname": "Arby's",
		"addr": "12700 Spine Road SW E-F5",
		"addr2": "Atlanta GA 30320",
		"phone": "7709533300"
	}, {
		"id": "2",
		"custid": "WILMAC ATBRAP ",
		"custname": "Atlanta Bread Airport",
		"addr": "12700 Spine Road Concourse A",
		"addr2": "Atlanta GA 30320"
	}, {
		"id": "3",
		"custid": "WILMAC ATLCHO ",
		"custname": "Atlanta Chop House",
		"addr": "6000 North Terminal F-8 Atrium",
		"addr2": "Atlanta GA 30230"
	}]
}
```
One should use the column name as the property name in the items of the JSON data. The format specify the property **independent on the order**. It simplifies the maintenance of the application because you can easy include and additional property in the items of data or to change the order of columns.

You will see the resulting grid on [**the demo**](http://www.ok-soft-gmbh.com/jqGrid/OK/actionsColumn1.html).

I removed in the demo additionally unneeded options `gridview:true, sortorder: "asc"` with default values and the default property `width: 150` for the last columns. I removed unneeded hidden column `id` too. The `id` property of the JSON data will be used to assign `id` attribute of the rows (`<tr>` elements). See [the picture](https://free-jqgrid.github.io/getting-started/index.html#grid-internal-div) for more details. I moved the information from `colNames` to `lable` of `colModel`. Finally I added `pagerRightWidth: 150` option to reduce the width of the right part of the page and to get more place for navigator icons. 

Alternatively you can hold the original format of the JSON data, but to use `jsonReader: { repeatitems: false }` option and `jsonmap` property for all columns, which read the input data:
```JavaScript
colModel: [
    { name: "act", template: "actions" },
    { name: "id", jsonmap: "cell.0", align: "center", width: 40, hidden: true },
    { name: "custid", jsonmap: "cell.1", width: 250, editable: false },
    { name: "custname", jsonmap: "cell.2", width: 250 },
    { name: "addr", jsonmap: "cell.3", width: 200, search: false },
    { name: "addr2", jsonmap: "cell.4", search: false },
    { name: "phone", jsonmap: "cell.5" }
]
```
See [**the corresponding demo**](http://www.ok-soft-gmbh.com/jqGrid/OK/actionsColumn.html).

I forgot to mention, that I used the latest versions of all dependent CSS and JavaScript components, with the exception of jQuery. I used jQuery 3.1.1 instead of jQuery 3.2.0, which is buggy in calculation of `width` and `outerWidth` (see [the issue](https://github.com/jquery/jquery/issues/3571)). In any way I'd recommend you to use free jqGrid 4.14.0 instead of old 4.13.1, which you used in your original demo.

Finally, I'd recommend you to consider to use `loadonce: true` and `forceClientSorting: true` options of free jqGrid and to return just an array of all items from the server, instead of implementing *server side paging*. The total number of items, which you have currently is very small: only 57 items. Free jqGrid implement local paging, sorting and filtering of data relatively effective. Try [the demo](http://www.ok-soft-gmbh.com/jqGrid/OK/performane-15-60000-25-free-jqgrid.htm) as an example. It has 60000 rows, but it uses local paging and displays only 25 rows

**Comment by efaux** [comment]
Thank you very much it working fine now.

**Comment by OlegKi** [comment]
You are welcome! I'm closing the issue.

**Comment by OlegKi** [comment]
@efaux: Short remark: jQuery 3.2.1 is published now and it can be used with free jqGrid (see [the modified demo](http://www.ok-soft-gmbh.com/jqGrid/OK/actionsColumn1.html), which uses now jQuery 3.2.1). [The bug](https://github.com/jquery/jquery/issues/3571) of 3.2.0, which I mention in the previous comment is fixed in 3.2.1.
